### PR TITLE
User guide for installing Openshift Routes with AS3 backend

### PR DESCRIPTION
**Pull request by ssurenr**
Reviewer: @kkinch

### CODE REVIEW

File: docs/openshift/kctlr-use-as3-backend.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+Once AS3 package is installed in BIG-IP, restart the controller with the following command line parameter
+`--agent=as3`. A sample controller deployment is provided below:
+
+.. code-block:: YAML
+   :emphasize-lines: 9
+   args: [
+         "--bigip-username=$(BIGIP_USERNAME)",
+         "--bigip-password=$(BIGIP_PASSWORD)",
+         "--bigip-url=10.10.10.10",
```
Comment (trinaths) [inline docs/openshift/kctlr-use-as3-backend.rst]: --bigip-url="<BIGIP  URL>" ?? just to remove any ambiguity

File: docs/openshift/kctlr-use-as3-backend.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+         ]
+
++---------------------+---------------+---------+--------------------------------------------------------------+
+| Argument            | Values        | Default |                      Description                             |
++=====================+===============+=========+==============================================================+
+| --agent             | as3, cccl     | cccl    | If as3 is given CIS creates one additional partition with    |
+|                     |               |         | name <partition>_AS3 where <partition> is provided through   |
+|                     |               |         | argument --bigip-paritition                                  |
++---------------------+---------------+---------+--------------------------------------------------------------+
+
```
Comment (trinaths) [inline docs/openshift/kctlr-use-as3-backend.rst]: May be we need to some info on --manage-ingress option too. May not be in this doc.

File: docs/openshift/kctlr-use-as3-backend.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+users manually.
+
+Supported Openshift Route Features
+----------------------------------
++-------+---------------+-------------------------------+---------------------------+
+| Route |  Termination  |          Option               |          Values           |
++=======+===============+===============================+===========================+
+| Host  |               |                               |                           |
++-------+---------------+-------------------------------+---------------------------+
+| Path  |               |                               |                           |
```
Comment (trinaths) [inline docs/openshift/kctlr-use-as3-backend.rst]: Port ?

File: docs/openshift/kctlr-use-as3-backend.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
++-------+---------------+-------------------------------+---------------------------+
+|       | - Reencrypt   | insecureEdgeTerminatoinPolicy | - None                    |
++-------+---------------+-------------------------------+---------------------------+
+
+**Annotations**
+- virtual-server.f5.com/balance
+- virtual-server.f5.com/clientssl
+- virtual-server.f5.com/serverssl
+- virtual-server.f5.com/secure-serverssl
+- virtual-server.f5.com/balance
```
Comment (trinaths) [inline docs/openshift/kctlr-use-as3-backend.rst]: duplicate. its mentioned above too

File: docs/openshift/kctlr-use-as3-backend.rst (hunk 1 above)
Comment (ssurenr) [reply docs/openshift/kctlr-use-as3-backend.rst]: Previous docs follow this. Leaving it to Kevin's decision.

File: docs/openshift/kctlr-use-as3-backend.rst (hunk 2 above)
Comment (ssurenr) [reply docs/openshift/kctlr-use-as3-backend.rst]: That is out of the scope of this doc.

File: docs/openshift/kctlr-use-as3-backend.rst (hunk 4 above)
Comment (ssurenr) [inline docs/openshift/kctlr-use-as3-backend.rst]:
```suggestion
```

File: docs/openshift/kctlr-use-as3-backend.rst (hunk 3 above)
Comment (ssurenr) [reply docs/openshift/kctlr-use-as3-backend.rst]: TargetPort is not a supported feature.

**Comment by ssurenr** [comment]
@kkinch https://clouddocs.f5networks.net/containers/latest/openshift/kctlr-use-as3-backend.html
looks excellent. Under "supported-openshift-route-features" section, the annotations should come as a list. All other this are good.

**Comment by kkinch** [comment]
@ssurenr Thank you for your new document, very well written and helpful! 

As I understand we're using my modified version. We will link to the new from here:

   https://clouddocs.f5networks.net/containers/v2/openshift/

   https://clouddocs.f5networks.net/containers/v2/openshift/kctlr-use-as3-backend.html 

If this is acceptable, I assume we can close this PR and I'll publish to livesite after go/no-go today.

**Comment by ssurenr** [comment]
@kkinch Sure. We can go ahead and publish your version. Could you please take care of the following:
1. Highlight `"--agent=AS3"` under **Enabling AS3 orchestration**.
2. **Annotations** should be a sub section of **Supported Openshift Route Features**.
3. **AS3 Resources** can be a main section; not listed under **Known Issues**.

**Comment by kkinch** [comment]
@ssurenr I have almost all updates. I think you're asking to hightlight --agent=as3 in the code block.

Currently, it's not bolding correctly. Will research this.

https://clouddocs.f5networks.net/containers/v2/openshift/kctlr-use-as3-backend.html

**Comment by ssurenr** [comment]
Sure. No problem. Thanks @kkinch .

**Pull request closed without merging**